src/js/utils/dom.js

```diff
@@ -857,3 +857,33 @@ export function computedStyle(el, prop) {
 
   return '';
 }
+
+/**
+ * Copy document style sheets to another window.
+ *
+ * @param    {Window} win
+ *           The window element you want to copy the document style sheets to.
+ *
+ */
+export function copyStyleSheetsToWindow(win) {
+  const allCSS = [...document.styleSheets]
+    .map((styleSheet) => {
+      try {
+        return [...styleSheet.cssRules].map((rule) => rule.cssText).join('');
+      } catch (e) {
+        const link = document.createElement('link');
```

Comment:
When would it be expected for code in the try block to fail so we reach here? Running locally and in CI the concatenation of styles works for both `<style>` and `<link>` els, hence the test failures - no `<link>` el is added to the doc.

If `<links>` did need to be added here, they would being added in the order they occur but the content of all `<style>`s would be in a `<style>` after those `<link>`s, potentially changing the original order of rules.

Reply:
Accessing `cssRules` from cross-origin `<link>` elements such as `<link href="https://cdn.com/styles.css" rel="stylesheet">` raise an error: "Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules". That's why we try/catch here and append them manually instead.

FWIW, tests pass locally.

Reply:
The reason the tests fail is that adding the the `<link>` stylesheet is async and it's not present in `document.stylesSheets` in time for the test.